### DISCUSSION TITLE
Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: Appium Settings app is not running after 5000ms  Host info: host: 'DESKTOP-PMUR4M5', ip: '192.168.1.2'

### DISCUSSION THREAD

**Discussion by faizmux** [appium/appium#18806] [Q&A]
public class BaseClass {
	
	public static AndroidDriver driver;
	
	@BeforeClass
	public void setup(String[] args) {
		try {
			UiAutomator2Options options = new UiAutomator2Options();
	        options.setDeviceName("e042712d");
	        options.setAutomationName("UiAutomator2");
	        options.setPlatformName("Android");
	        options.setPlatformVersion("13");
	        options.setAppPackage("com.educationfund.edufund");
	        options.setAppActivity("com.educationfund.MainActivity");
	        options.setIgnoreHiddenApiPolicyError(true);
	        options.setNoReset(true);
	        
			URL url = new URL("http://127.0.0.1:4723/wd/hub");
			
			driver = new AndroidDriver(url, options);
			Thread.sleep(5000);
			
		} catch(Exception exp) {
			System.out.println("Cause is : " +exp.getCause());
			System.out.println("Message is : "+exp.getMessage());
			exp.printStackTrace();
		}
		
	}

**Comment by faizmux** [comment]
[debug] [AppiumDriver@84ad] Calling AppiumDriver.createSession() with args: [null,null,{"firstMatch":[{}],"alwaysMatch":{"appium:appActivity":"com.educationfund.MainActivity","appium:appPackage":"com.educationfund.edufund","appium:automationName":"UiAutomator2","appium:deviceName":"e042712d","appium:ignoreHiddenApiPolicyError":true,"appium:noReset":true,"appium:platformVersion":"13","platformName":"ANDROID"}}]
[debug] [AppiumDriver@84ad] Event 'newSessionRequested' logged at 1687761409669 (12:06:49 GMT+0530 (India Standard Time))
[Appium] Attempting to find matching driver for automationName 'UiAutomator2' and platformName 'ANDROID'[Appium] The 'uiautomator2' driver was installed and matched caps.
[Appium] Will require it at C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver
[debug] [Appium] Requiring driver at C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver
[AppiumDriver@84ad] Appium v2.0.0-rc.3 creating new AndroidUiautomator2Driver (v2.27.0) session
[AppiumDriver@84ad] Checking BaseDriver versions for Appium and AndroidUiautomator2Driver
[AppiumDriver@84ad] Appium's BaseDriver version is 9.3.13
[AppiumDriver@84ad] AndroidUiautomator2Driver's BaseDriver version is 9.3.13
[debug] [AndroidUiautomator2Driver@2bcf] Creating session with W3C capabilities: {
[debug] [AndroidUiautomator2Driver@2bcf]   "alwaysMatch": {
[debug] [AndroidUiautomator2Driver@2bcf]     "platformName": "ANDROID",
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:appActivity": "com.educationfund.MainActivity",
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:appPackage": "com.educationfund.edufund",
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:automationName": "UiAutomator2",
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:deviceName": "e042712d",
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:ignoreHiddenApiPolicyError": true,
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:noReset": true,
[debug] [AndroidUiautomator2Driver@2bcf]     "appium:platformVersion": "13"
[debug] [AndroidUiautomator2Driver@2bcf]   },
[debug] [AndroidUiautomator2Driver@2bcf]   "firstMatch": [
[debug] [AndroidUiautomator2Driver@2bcf]     {}
[debug] [AndroidUiautomator2Driver@2bcf]   ]
[debug] [AndroidUiautomator2Driver@2bcf] }
[AndroidUiautomator2Driver@2bcf (a3bcd47b)] Session created with session id: a3bcd47b-da55-4366-bee5-278ea9f50e88
[AndroidUiautomator2Driver@2bcf (a3bcd47b)] Starting 'com.educationfund.edufund' directly on the device
[debug] [ADB] Running 'adb -P 5037 start-server'
[ADB] Using 'adb.exe' from 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe'
[AndroidDriver] Retrieving device list
[debug] [ADB] Trying to find a connected android device
[debug] [ADB] Getting connected devices
[debug] [ADB] Connected devices: [{"udid":"e042712d","state":"device"}]
[AndroidDriver] Looking for a device with Android '13.0.0'
[debug] [ADB] Setting device id to e042712d
[ADB] Getting device platform version
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell getprop ro.build.version.release'
[debug] [ADB] Current device property 'ro.build.version.release': 13
[AndroidDriver] Using device: e042712d
[debug] [ADB] Running 'adb -P 5037 start-server'
[ADB] Using 'adb.exe' from 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe'
[debug] [ADB] Setting device id to e042712d
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell getprop ro.build.version.sdk'
[debug] [ADB] Current device property 'ro.build.version.sdk': 33
[ADB] Getting device platform version
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell getprop ro.build.version.release'
[debug] [ADB] Current device property 'ro.build.version.release': 13
[debug] [ADB] Device API level: 33
[AndroidUiautomator2Driver@2bcf (a3bcd47b)] Relaxing hidden api policy
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell 'settings put global hidden_api_policy_pre_p_apps 1;settings put global hidden_api_policy_p_apps 1;settings put global hidden_api_policy 1''
[ADB] Failed to set setting keys 'hidden_api_policy_pre_p_apps,hidden_api_policy_p_apps,hidden_api_policy' to '1'. Original error: Error executing adbExec. Original error: 'Command 'C:\\Users\\faizm\\AppData\\Local\\Android\\sdk\\platform-tools\\adb.exe -P 5037 -s e042712d shell 'settings put global hidden_api_policy_pre_p_apps 1;settings put global hidden_api_policy_p_apps 1;settings put global hidden_api_policy 1'' exited with code 255'; Command output:
[ADB] Exception occurred while executing 'put':
[ADB] java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_SECURE_SETTINGS
[ADB]   at com.android.providers.settings.SettingsProvider.enforceWritePermission(SettingsProvider.java:2562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1562)
[ADB]   at com.android.providers.settings.SettingsProvider.insertGlobalSetting(SettingsProvider.java:1516)
[ADB]   at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:495)
[ADB]   at android.content.ContentProvider.call(ContentProvider.java:2511)
[ADB]   at android.content.ContentProvider$Transport.call(ContentProvider.java:525)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.putForUser(SettingsService.java:384)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.onCommand(SettingsService.java:280)
[ADB]   at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[ADB]   at android.os.ShellCommand.exec(ShellCommand.java:38)
[ADB]   at com.android.providers.settings.SettingsService.onShellCommand(SettingsService.java:51)
[ADB]   at android.os.Binder.shellCommand(Binder.java:1073)
[ADB]   at android.os.Binder.onTransact(Binder.java:901)
[ADB]   at android.os.Binder.execTransactInternal(Binder.java:1331)
[ADB]   at android.os.Binder.execTransact(Binder.java:1268)
[ADB]
[ADB] Exception occurred while executing 'put':
[ADB] java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_SECURE_SETTINGS
[ADB]   at com.android.providers.settings.SettingsProvider.enforceWritePermission(SettingsProvider.java:2562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1562)
[ADB]   at com.android.providers.settings.SettingsProvider.insertGlobalSetting(SettingsProvider.java:1516)
[ADB]   at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:495)
[ADB]   at android.content.ContentProvider.call(ContentProvider.java:2511)
[ADB]   at android.content.ContentProvider$Transport.call(ContentProvider.java:525)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.putForUser(SettingsService.java:384)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.onCommand(SettingsService.java:280)
[ADB]   at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[ADB]   at android.os.ShellCommand.exec(ShellCommand.java:38)
[ADB]   at com.android.providers.settings.SettingsService.onShellCommand(SettingsService.java:51)
[ADB]   at android.os.Binder.shellCommand(Binder.java:1073)
[ADB]   at android.os.Binder.onTransact(Binder.java:901)
[ADB]   at android.os.Binder.execTransactInternal(Binder.java:1331)
[ADB]   at android.os.Binder.execTransact(Binder.java:1268)
[ADB]
[ADB] Exception occurred while executing 'put':
[ADB] java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_SECURE_SETTINGS
[ADB]   at com.android.providers.settings.SettingsProvider.enforceWritePermission(SettingsProvider.java:2562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1562)
[ADB]   at com.android.providers.settings.SettingsProvider.insertGlobalSetting(SettingsProvider.java:1516)
[ADB]   at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:495)
[ADB]   at android.content.ContentProvider.call(ContentProvider.java:2511)
[ADB]   at android.content.ContentProvider$Transport.call(ContentProvider.java:525)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.putForUser(SettingsService.java:384)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.onCommand(SettingsService.java:280)
[ADB]   at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[ADB]   at android.os.ShellCommand.exec(ShellCommand.java:38)
[ADB]   at com.android.providers.settings.SettingsService.onShellCommand(SettingsService.java:51)
[ADB]   at android.os.Binder.shellCommand(Binder.java:1073)
[ADB]   at android.os.Binder.onTransact(Binder.java:901)
[ADB]   at android.os.Binder.execTransactInternal(Binder.java:1331)
[ADB]   at android.os.Binder.execTransact(Binder.java:1268)
[ADB]
[AndroidDriver] No app sent in, not parsing package/activity
[debug] [AndroidDriver] Pushing settings apk to device...
[debug] [ADB] Getting install status for io.appium.settings
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell dumpsys package io.appium.settings'
[debug] [ADB] 'io.appium.settings' is installed
[debug] [ADB] Getting package info for 'io.appium.settings'
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell dumpsys package io.appium.settings'
[debug] [ADB] The version name of the installed 'io.appium.settings' is greater or equal to the application version name ('5.0.4' >= '5.0.4')
[debug] [ADB] There is no need to install/upgrade 'C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\io.appium.settings\apks\settings_apk-debug.apk'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell 'pgrep --help; echo $?''
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell appops set io.appium.settings PROJECT_MEDIA allow'
[debug] [AndroidDriver] Error executing adbExec. Original error: 'Command 'C:\\Users\\faizm\\AppData\\Local\\Android\\sdk\\platform-tools\\adb.exe -P 5037 -s e042712d shell appops set io.appium.settings PROJECT_MEDIA allow' exited with code 255'; Command output:
[debug] [AndroidDriver] Exception occurred while executing 'set':
[debug] [AndroidDriver] java.lang.SecurityException: uid 2000 does not have android.permission.MANAGE_APP_OPS_MODES.
[debug] [AndroidDriver]         at android.app.ContextImpl.enforce(ContextImpl.java:2253)
[debug] [AndroidDriver]         at android.app.ContextImpl.enforcePermission(ContextImpl.java:2262)
[debug] [AndroidDriver]         at com.android.server.appop.AppOpsService.enforceManageAppOpsModes(AppOpsService.java:2586)
[debug] [AndroidDriver]         at com.android.server.appop.AppOpsService.setMode(AppOpsService.java:2855)
[debug] [AndroidDriver]         at com.android.server.appop.AppOpsService.setMode(AppOpsService.java:2850)
[debug] [AndroidDriver]         at com.android.server.appop.AppOpsService.onShellCommand(AppOpsService.java:5680)
[debug] [AndroidDriver]         at com.android.server.appop.AppOpsService$Shell.onCommand(AppOpsService.java:5449)
[debug] [AndroidDriver]         at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[debug] [AndroidDriver]         at android.os.ShellCommand.exec(ShellCommand.java:38)
[debug] [AndroidDriver]         at com.android.server.appop.AppOpsService.onShellCommand(AppOpsService.java:5621)
[debug] [AndroidDriver]         at android.os.Binder.shellCommand(Binder.java:1073)
[debug] [AndroidDriver]         at android.os.Binder.onTransact(Binder.java:901)
[debug] [AndroidDriver]         at com.android.internal.app.IAppOpsService$Stub.onTransact(IAppOpsService.java:1120)
[debug] [AndroidDriver]         at android.os.Binder.execTransactInternal(Binder.java:1331)
[debug] [AndroidDriver]         at android.os.Binder.execTransact(Binder.java:1268)
[debug] [AndroidDriver]
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell cmd notification allow_listener io.appium.settings/.NLService'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Starting Appium Settings app
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell am start -n io.appium.settings/.Settings -a android.intent.action.MAIN -c android.intent.category.LAUNCHER'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [AndroidDriver] Error: Appium Settings app is not running after 5000ms
[debug] [AndroidDriver]     at ADB.requireRunningSettingsApp (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-adb\lib\tools\settings-client-commands.js:100:11)
[debug] [AndroidDriver]     at Object.pushSettingsApp (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\android-helpers.js:677:5)
[debug] [AndroidDriver]     at Object.initDevice (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\android-helpers.js:823:5)
[debug] [AndroidDriver]     at AndroidUiautomator2Driver.startUiAutomator2Session (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\lib\driver.js:393:5)
[debug] [AndroidDriver]     at AndroidUiautomator2Driver.createSession (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\lib\driver.js:241:7)
[debug] [AndroidDriver]     at AppiumDriver.createSession (C:\Users\faizm\AppData\Roaming\npm\node_modules\appium\lib\appium.js:346:35)
[debug] [AndroidUiautomator2Driver@2bcf (a3bcd47b)] Deleting UiAutomator2 session
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell dumpsys activity services io.appium.settings/.recorder.RecorderService'
[AndroidUiautomator2Driver@2bcf (a3bcd47b)] Restoring hidden api policy to the device default configuration
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy''
[ADB] Failed to delete keys 'hidden_api_policy_pre_p_apps,hidden_api_policy_p_apps,hidden_api_policy'. Original error: Error executing adbExec. Original error: 'Command 'C:\\Users\\faizm\\AppData\\Local\\Android\\sdk\\platform-tools\\adb.exe -P 5037 -s e042712d shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy'' exited with code 255'; Command output:
[ADB] Exception occurred while executing 'delete':
[ADB] java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_SECURE_SETTINGS
[ADB]   at com.android.providers.settings.SettingsProvider.enforceWritePermission(SettingsProvider.java:2562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1554)
[ADB]   at com.android.providers.settings.SettingsProvider.deleteGlobalSetting(SettingsProvider.java:1525)
[ADB]   at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:568)
[ADB]   at android.content.ContentProvider.call(ContentProvider.java:2511)
[ADB]   at android.content.ContentProvider$Transport.call(ContentProvider.java:525)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.deleteForUser(SettingsService.java:410)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.onCommand(SettingsService.java:284)
[ADB]   at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[ADB]   at android.os.ShellCommand.exec(ShellCommand.java:38)
[ADB]   at com.android.providers.settings.SettingsService.onShellCommand(SettingsService.java:51)
[ADB]   at android.os.Binder.shellCommand(Binder.java:1073)
[ADB]   at android.os.Binder.onTransact(Binder.java:901)
[ADB]   at android.os.Binder.execTransactInternal(Binder.java:1331)
[ADB]   at android.os.Binder.execTransact(Binder.java:1268)
[ADB]
[ADB] Exception occurred while executing 'delete':
[ADB] java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_SECURE_SETTINGS
[ADB]   at com.android.providers.settings.SettingsProvider.enforceWritePermission(SettingsProvider.java:2562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1554)
[ADB]   at com.android.providers.settings.SettingsProvider.deleteGlobalSetting(SettingsProvider.java:1525)
[ADB]   at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:568)
[ADB]   at android.content.ContentProvider.call(ContentProvider.java:2511)
[ADB]   at android.content.ContentProvider$Transport.call(ContentProvider.java:525)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.deleteForUser(SettingsService.java:410)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.onCommand(SettingsService.java:284)
[ADB]   at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[ADB]   at android.os.ShellCommand.exec(ShellCommand.java:38)
[ADB]   at com.android.providers.settings.SettingsService.onShellCommand(SettingsService.java:51)
[ADB]   at android.os.Binder.shellCommand(Binder.java:1073)
[ADB]   at android.os.Binder.onTransact(Binder.java:901)
[ADB]   at android.os.Binder.execTransactInternal(Binder.java:1331)
[ADB]   at android.os.Binder.execTransact(Binder.java:1268)
[ADB]
[ADB] Exception occurred while executing 'delete':
[ADB] java.lang.SecurityException: Permission denial: writing to settings requires:android.permission.WRITE_SECURE_SETTINGS
[ADB]   at com.android.providers.settings.SettingsProvider.enforceWritePermission(SettingsProvider.java:2562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1562)
[ADB]   at com.android.providers.settings.SettingsProvider.mutateGlobalSetting(SettingsProvider.java:1554)
[ADB]   at com.android.providers.settings.SettingsProvider.deleteGlobalSetting(SettingsProvider.java:1525)
[ADB]   at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:568)
[ADB]   at android.content.ContentProvider.call(ContentProvider.java:2511)
[ADB]   at android.content.ContentProvider$Transport.call(ContentProvider.java:525)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.deleteForUser(SettingsService.java:410)
[ADB]   at com.android.providers.settings.SettingsService$MyShellCommand.onCommand(SettingsService.java:284)
[ADB]   at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
[ADB]   at android.os.ShellCommand.exec(ShellCommand.java:38)
[ADB]   at com.android.providers.settings.SettingsService.onShellCommand(SettingsService.java:51)
[ADB]   at android.os.Binder.shellCommand(Binder.java:1073)
[ADB]   at android.os.Binder.onTransact(Binder.java:901)
[ADB]   at android.os.Binder.execTransactInternal(Binder.java:1331)
[ADB]   at android.os.Binder.execTransact(Binder.java:1268)
[ADB]
[debug] [AppiumDriver@84ad] Event 'newSessionStarted' logged at 1687761417783 (12:06:57 GMT+0530 (India Standard Time))
[debug] [AppiumDriver@84ad] Encountered internal error running command: Error: Appium Settings app is not running after 5000ms
[debug] [AppiumDriver@84ad]     at ADB.requireRunningSettingsApp (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-adb\lib\tools\settings-client-commands.js:100:11)
[debug] [AppiumDriver@84ad]     at Object.pushSettingsApp (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\android-helpers.js:677:5)
[debug] [AppiumDriver@84ad]     at Object.initDevice (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-android-driver\lib\android-helpers.js:823:5)
[debug] [AppiumDriver@84ad]     at AndroidUiautomator2Driver.startUiAutomator2Session (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\lib\driver.js:393:5)
[debug] [AppiumDriver@84ad]     at AndroidUiautomator2Driver.createSession (C:\Users\faizm\.appium\node_modules\appium-uiautomator2-driver\lib\driver.js:241:7)
[debug] [AppiumDriver@84ad]     at AppiumDriver.createSession (C:\Users\faizm\AppData\Roaming\npm\node_modules\appium\lib\appium.js:346:35)
[HTTP] <-- POST /wd/hub/session 500 8143 ms - 660

**Comment by KazuCocoa** [comment]
It seems like the primary reason was `io.appium.settings` failed to start on your device.

What happens in the logcat when you run adb command below to start the settings app?

```
[debug] [ADB] Running 'C:\Users\faizm\AppData\Local\Android\sdk\platform-tools\adb.exe -P 5037 -s e042712d shell am start -n io.appium.settings/.Settings -a android.intent.action.MAIN -c android.intent.category.LAUNCHER'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
```
